### PR TITLE
Selenium devops

### LINKED ISSUE #818: Setup BrowserStack in AzureDevOps
Now that BrowserStack open source account is set up, use those credentials in the azure devops pipeline

### PR DESCRIPTION
**Pull request by cailyncodes**
Should close #818 

Incorporates the ability to use BrowserStack (both on a public url [e.g. staging] and a private url [e.g. local developer machine]).
 
Updates the pipeline to publish uncompiled ClientApp source code to run E2E tests.

I added a stage to the staging release pipeline to run E2E tests. I currently have it set to manual only (we can connect it as a subsequent stage to our release once this lands in master).

Currently, Azure Devops is using my personal browserstack account and credentials -- so will need to change that.

Possible TODOs: Add E2E during the build stage (we could do this by running the client in a docker container and updating the docker-compose.test.yml file); publish test results on BrowserStack; publish test results to Azure DevOps (this might already be handled?); update other release pipelines.

CC @cmgiven -- we will need to create an organizational account in BrowserStack and request free account with open source URL

Other notes:
There are four possible running configurations:
* 1) Private URL + Selenium Grid (local developer testing with docker containers for browsers)
* 2) Public URL + Selenium Grid (we don't use -- would be local developer testing staging)
* 3) Private URL + BrowserStack (local developer testing with browserstack browsers OR azure deveops build pipeline testing with browserstack browsers)
* 4) Public URL + BrowserStack (Azure Devops Release pipeline testing with browserstack browsers)

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -165,15 +172,15 @@ jobs:
...
+    # condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/heads/master'))
     inputs:
       pathToPublish: "$(Build.ArtifactStagingDirectory)/hedwig-spa.zip"
       artifactName: hedwig-spa
 
   # Publish artifact to dropbox - hedwig-db
   - task: PublishBuildArtifacts@1
     displayName: "Publish Artifacts: hedwig-db"
-    condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/heads/master'))
+    # condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/heads/master'))
```
Comment (jlhogan) [inline azure-pipelines.yml]: should this be checked in ?

**Comment by jlhogan** [comment]
sweeeeeeeeeet. would say :heavy_plus_sign: to adding to build stage if these run in < a few mins in the pipeline

File: azure-pipelines.yml (hunk 1 above)
Comment (cailyncodes) [reply azure-pipelines.yml]: No -- it's just there for me to test while it's on a PR branch

**Comment by cailyncodes** [comment]
On hold pending license update

**Pull request merged**